qsimcirq/qsim_simulator.py

```diff
@@ -62,6 +64,120 @@ def __init__(self, qsim_options: dict = {}):
     self.qsim_options.update(qsim_options)
     return
 
+  def _run(
+        self,
+        circuit: circuits.Circuit,
+        param_resolver: study.ParamResolver,
+        repetitions: int
+  ) -> Dict[str, np.ndarray]:
+    """Run a simulation, mimicking quantum hardware.
+
+    Args:
+        program: The circuit to simulate.
+        param_resolver: Parameters to run with the program.
+        repetitions: Number of times to repeat the run.
+
+    Returns:
+        A dictionary from measurement gate key to measurement
+        results.
+    """
+    param_resolver = param_resolver or study.ParamResolver({})
+    solved_circuit = protocols.resolve_parameters(circuit, param_resolver)
+
+    return self._sample_bitstrings(solved_circuit, repetitions)
+
+  def _sample_bitstrings(
+    self,
+    program: circuits.Circuit,
+    repetitions: int = 1,
+  ) -> Dict[str, np.ndarray]:
+    """Samples bitstrings from the circuit
+
+    All MeasurementGates must be terminal.
+    Note that this does not collapse the wave function.
```

Comment:
We actually collapse the wave function. Is that consistent with what Cirq expects?

Reply:
Removed this comment. Cirq does expect intermediate measurements to collapse the wave function.